crates/key-value/src/lib.rs

```diff
@@ -138,6 +139,100 @@ impl key_value::HostStore for KeyValueDispatch {
     }
 }
 
+fn to_wasi_err(e: Error) -> wasi_keyvalue::store::Error {
+    match e {
+        Error::AccessDenied => wasi_keyvalue::store::Error::AccessDenied,
+        Error::NoSuchStore => wasi_keyvalue::store::Error::NoSuchStore,
+        Error::StoreTableFull => wasi_keyvalue::store::Error::Other("store table full".to_string()),
+        Error::Other(msg) => wasi_keyvalue::store::Error::Other(msg),
+    }
+}
+
+#[async_trait]
+impl wasi_keyvalue::store::Host for KeyValueDispatch {
+    async fn open(
+        &mut self,
+        identifier: String,
+    ) -> anyhow::Result<Result<Resource<wasi_keyvalue::store::Bucket>, wasi_keyvalue::store::Error>>
```

Comment:
You can use the [`trappable_error_type`](https://github.com/fermyon/spin/blob/13a133fded7c48fa5d1931207619379e737e9376/crates/world/src/lib.rs#L14-L28) option to make the signatures here a bit easier to work with. See the `llm` implementation for an example use.

Reply:
Thanks @rylev.  I gave this a try and am not sure how I feel about.  I've left it in a separate commit and would value your thoughts...